HelloWorld.cs

```diff
@@ -0,0 +1,345 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections;
+using System.Diagnostics;
+
+internal class Program
+{
+    private static void Main(string[] args)
+    {
+        string osDetect = DetectOS();
+
+        if (args.Length >= 1)
+        {
+            osDetect = args[0];
+        }
+
+        if (osDetect == "linux")
+        {
+            DrawLinux();
+        }
+        else if (osDetect == "freebsd")
+        {
+            DrawFreeBSD();
+        }
+        else if (osDetect == "mac")
+        {
+            DrawMac();
+        }
+        else if (osDetect == "windows")
+        {
+            DrawWindows();
+        }
+        else
+        {
+            DrawWindows();
+        }
+
+        Console.WriteLine();
+        Console.WriteLine("Press ENTER to continue . . . ");
+        Console.ReadLine();
+    }
+
+    private static void DrawWindows()
+    {
+        Console.WriteLine("Hello, Windows...");
+
+        const int squareSize = 20;
+
+        var colors = new[] { ConsoleColor.Red, ConsoleColor.Green, ConsoleColor.Blue, ConsoleColor.Yellow };
+        for (int row = 0; row < 2; row++)
+        {
+            for (int i = 0; i < squareSize / 2; i++)
+            {
+                Console.WriteLine();
+                Console.Write("  ");
+                for (int col = 0; col < 2; col++)
+                {
+                    Console.BackgroundColor = colors[row * 2 + col];
+                    Console.ForegroundColor = colors[row * 2 + col];
+                    for (int j = 0; j < squareSize; j++) Console.Write('@');
+                    Console.ResetColor();
+                }
+            }
+        }
+        Console.WriteLine();
+    }
+
```

Comment:
I didn’t see a place to create a feature branch, but did see an option to submit a file in the fork dialog. Please advise if change is needed. Summary is addition of new function, and minor modification of conditionals in Main. No other changes.

From: Krzysztof Cwalina [mailto:notifications@github.com]
Sent: Tuesday, June 23, 2015 2:01 PM
To: dotnet/corefxlab
Cc: Ryan Elisei
Subject: Re: [corefxlab] Update to HelloWorld.cs (#107)


In HelloWorld.cs<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_dotnet_corefxlab_pull_107-23discussion-5Fr33091816&d=AAMCaQ&c=fa_WZs7nNMvOIDyLmzi2sMVHyyC4hN9WQl29lWJQ5Y4&r=y-0K84a9Ub5AD0N82nRYuVi5GyfK1ArLuB4G5qhlOC0&m=T_hA6Nk4WTdvrOrHlbeCgzHtcDfOX9iOsMN9rreO6ow&s=XrizN0Hwdz6Q6drMTSCA_2y9jzjpapQPx9quHRDAS3o&e=>:

> +            for (int i = 0; i < squareSize / 2; i++)

> +            {

> +                Console.WriteLine();

> +                Console.Write("  ");

> +                for (int col = 0; col < 2; col++)

> +                {

> +                    Console.BackgroundColor = colors[row * 2 + col];

> +                    Console.ForegroundColor = colors[row * 2 + col];

> +                    for (int j = 0; j < squareSize; j++) Console.Write('@');

> +                    Console.ResetColor();

> +                }

> +            }

> +        }

> +        Console.WriteLine();

> +    }

> +

Why all these show as new? Did you change anything in the Draw* methods?

—
Reply to this email directly or view it on GitHub<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_dotnet_corefxlab_pull_107_files-23r33091816&d=AAMCaQ&c=fa_WZs7nNMvOIDyLmzi2sMVHyyC4hN9WQl29lWJQ5Y4&r=y-0K84a9Ub5AD0N82nRYuVi5GyfK1ArLuB4G5qhlOC0&m=T_hA6Nk4WTdvrOrHlbeCgzHtcDfOX9iOsMN9rreO6ow&s=OcRu2JGKtUJWC9L6tvGP7LNoeO61bt705D4438y-rfg&e=>.